src/lib/tasks/chunk_migration_task.cpp

```diff
@@ -0,0 +1,66 @@
+#include "chunk_migration_task.hpp"
+
+#if OPOSSUM_NUMA_SUPPORT
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "storage/numa_placement_manager.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+ChunkMigrationTask::ChunkMigrationTask(const std::string& table_name, const ChunkID chunk_id, int node_id)
+    : _table_name(table_name), _node_id(node_id), _chunk_ids({chunk_id}) {}
+
+ChunkMigrationTask::ChunkMigrationTask(const std::string& table_name, const std::vector<ChunkID>& chunk_ids,
+                                       int node_id)
+    : _table_name(table_name), _node_id(node_id), _chunk_ids(chunk_ids) {}
+
+void ChunkMigrationTask::_on_execute() {
+  auto table = StorageManager::get().get_table(_table_name);
+
+  if (!table) {
+    throw std::logic_error("Table does not exist.");
+  }
+
+  for (auto chunk_id : _chunk_ids) {
+    if (chunk_id >= table->chunk_count()) {
```

Comment:
`DebugAssert(chunk_id < table->chunk_count(), "Chunk with given ID does not exist.");`